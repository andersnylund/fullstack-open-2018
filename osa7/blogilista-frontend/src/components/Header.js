import React from 'react';
import PropTypes from 'prop-types';

import { Button, Grid, Typography, withStyles, } from '@material-ui/core';

const styles = {
  header: {
    textAlign: 'left',
  },
  info: {
    textAlign: 'right',
    display: 'flex',
    justifyContent: 'right',
  },
  infoItem: {
    padding: '10px 10px',
  },
};

const Header = (props) => {

  const { user, classes, onLogout, } = props;

  return (
    <Grid container spacing={24} className={classes.container}>
      <Grid item xs={6} className={classes.header}>
        <Typography variant='display3'>Blogs</Typography>
      </Grid>
      <Grid item xs={6}>
        {user !== null ?
          <div className={classes.info}>
            <Typography variant='subheading' className={classes.infoItem}><strong>{user.username}</strong> logged in</Typography>
            <Button size='small' variant='contained' onClick={onLogout} className={classes.infoItem}>Logout</Button>
          </div> :
          null}
      </Grid>
    </Grid>
  );
};

Header.propTypes = {
  user: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default withStyles(styles)(Header);

