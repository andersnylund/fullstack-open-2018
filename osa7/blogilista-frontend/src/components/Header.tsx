import React from 'react';

import { Button, Grid, Typography, withStyles, } from '@material-ui/core';

const styles: any = {
  header: {
    textAlign: 'left',
  },
  info: {
    textAlign: 'right',
    display: 'flex',
    justifyContent: 'right',
  },
  infoItem: {
    padding: '10px 10px',
  },
};

interface IProps {
  user: {
    username: string,
  },
  classes: {
    container: string,
    header: string,
    infoItem: string,
    info: string
  },
  onLogout: any
}


const Header: React.SFC<IProps> = ({ user, classes, onLogout, }) => {

  return (
    <Grid container={true} spacing={24} className={classes.container}>
      <Grid item={true} xs={6} className={classes.header}>
        <Typography variant='display3'>Blogs</Typography>
      </Grid>
      <Grid item={true} xs={6}>
        {user ?
          <div className={classes.info}>
            <Typography variant='subheading' className={classes.infoItem}><strong>{user.username}</strong> logged in</Typography>
            <Button size='small' variant='contained' onClick={onLogout} className={classes.infoItem}>Logout</Button>
          </div> :
          null}
      </Grid>
    </Grid>
  );
};

export default withStyles(styles)(Header);

