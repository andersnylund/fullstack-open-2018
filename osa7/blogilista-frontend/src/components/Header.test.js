import React from 'react';
import { shallow, mount, } from 'enzyme';
import Header from './Header';
import { Button, } from '@material-ui/core';

describe('<Header />', () => {

  const user = {
    username: 'test',
  };

  const handleLogout = jest.fn();

  it('shows login info when user is logged in', () => {
    const headerComponent = shallow(<Header user={user} onLogout={handleLogout}></Header>);
    expect(headerComponent.html()).toContain('logged in');
  });

  it('does not show login info when user is logged in', () => {
    const headerComponent = shallow(<Header user={undefined} onLogout={handleLogout}></Header>);
    expect(headerComponent.html()).not.toContain('logged in');
  });

  it('handles the logout', () => {
    const headerComponent = mount(<Header user={user} onLogout={handleLogout}></Header>);
    // console.log(headerComponent.debug());
    const button = headerComponent.find(Button);
    button.simulate('click');
    expect(handleLogout.mock.calls.length).toEqual(1);
  });

});