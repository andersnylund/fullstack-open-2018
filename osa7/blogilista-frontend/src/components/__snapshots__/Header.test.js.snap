// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<div
  className="MuiGrid-container-4 MuiGrid-spacing-xs-24-28"
>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-6-38 Header-header-1"
  >
    <h1
      className="MuiTypography-root-101 MuiTypography-display3-103"
    >
      Blogs
    </h1>
  </div>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-6-38"
  >
    <div
      className="Header-info-2"
    >
      <h3
        className="MuiTypography-root-101 MuiTypography-subheading-108 Header-infoItem-3"
      >
        <strong>
          test
        </strong>
         logged in
      </h3>
      <button
        className="MuiButtonBase-root-150 MuiButton-root-126 MuiButton-contained-135 MuiButton-raised-138 MuiButton-sizeSmall-147 Header-infoItem-3"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-127"
        >
          Logout
        </span>
        <span
          className="MuiTouchRipple-root-153"
        />
      </button>
    </div>
  </div>
</div>
`;
